File: backup/src/jclang/Diagnostic.java

package jclang;

import com.sun.jna.PointerType;
import org.jetbrains.annotations.NotNull;
import jclang.structs.CXString;
import jclang.util.Util;

public class Diagnostic extends PointerType {
    public enum DisplayOptions {
        DISPLAY_SOURCE_LOCATION,
        DISPLAY_COLUMN,
        DISPLAY_SOURCE_RANGES,
        DISPLAY_OPTION,
        DISPLAY_CATEGORY_ID,
        DISPLAY_CATEGORY_NAME
    }

    public enum Severity {
        IGNORED,
        NOTE,
        WARNING,
        ERROR,
        FATAL
    }

    @NotNull
    public String format(@NotNull DisplayOptions... options) {
        int flags = Util.buildOptionsMask(options);
        CXString.ByValue string = LibClang.I.formatDiagnostic(this, flags);
        NativePool.I.record(string);
        return LibClang.I.getCString(string);
    }

    @NotNull
    public Severity getSeverity() {
        int severity = LibClang.I.getDiagnosticSeverity(this);
        return Severity.values()[severity];
    }
}

File: libclang/src/main/java/klang/jvm/Diagnostic.kt
package klang.jvm

import com.sun.jna.PointerType

class Diagnostic : PointerType() {
    enum class DisplayOptions {
        DISPLAY_SOURCE_LOCATION,
        DISPLAY_COLUMN,
        DISPLAY_SOURCE_RANGES,
        DISPLAY_OPTION,
        DISPLAY_CATEGORY_ID,
        DISPLAY_CATEGORY_NAME
    }

    enum class Severity {
        IGNORED,
        NOTE,
        WARNING,
        ERROR,
        FATAL
    }

    fun format(vararg options: DisplayOptions): String {
        val flags = buildOptionsMask(*options)
        return Clang.formatDiagnostic(this, flags)
    }

    val severity: Severity
        get() {
            val severity = Clang.getDiagnosticSeverity(this)
            return Severity.values()[severity]
        }
}